### PR TITLE
Support for cluster autoscaling via config

### PR DESCRIPTION
**Pull request by jlowin**
This PR adds support for cluster autoscaling via config (`defaults.yaml`) or, by extension, CLI

### CODE REVIEW

File: dask_kubernetes/cli/defaults.yaml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 cluster:
-  num_nodes: 6
-  machine_type: n1-standard-4
+  num_nodes: 2
+  machine_type: n1-standard-8
```
Comment (martindurant) [inline dask_kubernetes/cli/defaults.yaml]: Why this change?

Comment (jlowin) [reply dask_kubernetes/cli/defaults.yaml]: GKE clusters have no fee for 5 or fewer nodes, but incur hourly charges above 5 (see https://cloud.google.com/container-engine/pricing), so I've gotten in the habit of using fewer numbers of larger instances for a set number of cores. I don't have a particular reason for those specific settings other than that they're the ones I tested with -- happy to set whatever you want! (though I do recommend a default <= 5)

**Comment by jlowin** [comment]
You know I am looking at the autoscaling docs (https://cloud.google.com/container-engine/docs/cluster-autoscaler) and I think the feature is a little different than described in the README. 

GKE autoscaling (of the cluster) simply looks at whether all pods can be scheduled, and adds/removes nodes appropriately (but doesn't change pods).

Kubernetes autoscaling (of a deployment or rc) looks at CPU usage and adds/removes pods appropriately (but doesn't change nodes)

So I think the setting affects cluster autoscaling but the README is describing Kubernetes autoscaling (which can be separately turned on/off via kubectl). Do you agree? I'm worried I am overlooking something.

File: dask_kubernetes/cli/defaults.yaml (hunk 1 above)
Comment (martindurant) [reply dask_kubernetes/cli/defaults.yaml]: I didn't know that... Let's go for 3 nodes of the bigger machines, then, so as not to reduce capacity.

**Comment by martindurant** [comment]
Probably what we want, is provide the option to have the cluster autoscale to fit pods (as you've done) and *not* mention CPU-based replication auto-scaling at all. The user has the ability to add pods if they wish. Then the rescale CLI command should indicate that there should be no need to specify the number of nodes if you are using auto-scaling.

**Comment by jlowin** [comment]
Just pushed an update to the readme and also the cli. It will show a warning if you try to resize with autoscaling enabled.

**Comment by martindurant** [comment]
Perfect!
One last question: the doc says that in theory, when a node is pruned, containers can be reorganised among the remaining ones, and can necessitate the restarting of some of them. Since the notebook and scheduler are first up, and any new nodes beyond the initial ones contain only additional worker containers, is it reasonable to assume that this will never happen to the notebook or scheduler containers? This I guess is not specific to auto-scaling.

**Comment by jlowin** [comment]
Great question. I think it's a reasonable assumption but I guess it's not guaranteed.

**Comment by martindurant** [comment]
OK, well we'll likely find out if it ever becomes a problem and change recommendations appropriately.
